Nomic_Analysis/1.Exploratory_Analysis/SHSY5Y/nbconverted/analysis_SHSY5Y.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from matplotlib.backends.backend_pdf import PdfPages
+from sklearn import preprocessing
+
+# In[2]:
+
+
+path = pathlib.Path("../../Data/clean/Plate2/nELISA_plate_430420_SH_SY5Y.csv")
+
+manual_cluster_1_path = pathlib.Path(
+    "../../Data/clean/Plate2/Manual_Treatment_Clusters_1.csv"
+)
+
+manual_cluster_2_path = pathlib.Path(
+    "../../Data/clean/Plate2/Manual_Treatment_Clusters_2.csv"
+)
+
+df = pd.read_csv(path)
+
+manual_clusters_1 = pd.read_csv(manual_cluster_1_path)
+manual_clusters_2 = pd.read_csv(manual_cluster_2_path)
+
+
+# In[3]:
+
+
+# select data only columns and make floats
+nELISA_data_values = df.filter(like="NSU", axis=1)
+nELISA_data_values = nELISA_data_values.astype("float")
+nELISA_data_values.head()
+
+
+# In[4]:
+
+
+# normalize data via max value in each column
+max_values = nELISA_data_values.max()  # find max value in each column
+nELISA_data_values_sensor_max_norm = nELISA_data_values.div(
+    max_values
+)  # divide each value in each column by max value in that column
+nELISA_data_values_sensor_max_norm.head()
+# min max normalization via sklearn
+
+
+# normalize data via min max normalization
+min_max_scaler = preprocessing.MinMaxScaler()
+nELISA_data_values_min_max_norm = min_max_scaler.fit_transform(nELISA_data_values)
+nELISA_data_values_min_max_norm = pd.DataFrame(
+    nELISA_data_values_min_max_norm, columns=nELISA_data_values.columns
+)
+nELISA_data_values_min_max_norm.head()
+
+
+# In[5]:
+
+
+# drop columns that are named with NSU
+Metadata = df.drop(df.filter(like="NSU", axis=1), axis=1)
+Metadata = Metadata.drop(df.filter(like="pgML", axis=1), axis=1)
+
+
+# In[6]:
+
+
+analysis_df = pd.concat([Metadata, nELISA_data_values_min_max_norm], axis=1)
+
+
+# In[7]:
+
+
+# add manual clusters columns to dataframe
+nELISA_plate_430420 = pd.merge(
+    analysis_df,
+    manual_clusters_2,
+    on=(
+        "inducer1",
+        "inducer1_concentration_value",
+        "inhibitor",
+        "inhibitor_concentration_value",
+        "inducer2",
+        "inducer2_concentration_value",
+    ),
+    how="inner",
+)
+
+
+# In[8]:
+
+
+# nELISA_plate_430420['treatment'] =
```

Comment:
Consider removing this line (it looks like it is unused code).

Reply:
removed!